### PR TITLE
luci-base/mod-network: add system cert bundle validation

### PR DESCRIPTION
**Pull request by swg0101**
luci-mod-network: add system cert bundle validation

This commit adds the ability for users to validate against the system's 
built-in CA bundle if it is installed. The process is made much easier because 
the user does not have to first extract the CA certificate from the EAPOL 
handshake and upload it via LuCI uploads. Dependent on commit 
openwrt/openwrt#2696.

luci-base: add system cert bundle detection

Add detection mechanism for system cert CA bundle installed by the 
ca-bundle package. Used by LuCI to detect whether the "Use system 
certificates" certificate validation option should be enabled.

Signed-off-by: David Lam <david@thedavid.net>

### CODE REVIEW

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 1]
```diff
@@ -1428,11 +1428,28 @@ return L.view.extend({
...
+					o = ss.taboption('encryption', form.Flag, 'ca_cert_usesystem', _('Use system certificates'), _("Validate server certificate using built-in system CA bundle,<br />requires the \"ca-bundle\" package"))
+					o.enabled = '1';
+					o.disabled = '0';
+					o.default = o.disabled;
 					o.depends({ mode: 'sta', encryption: 'wpa' });
 					o.depends({ mode: 'sta', encryption: 'wpa2' });
 					o.depends({ mode: 'sta-wds', encryption: 'wpa' });
 					o.depends({ mode: 'sta-wds', encryption: 'wpa2' });
+					o.validate = function(section_id, value) {
+						var eo = this.section.children.filter(function(o) { return o.option == 'ca_cert_usesystem' })[0],
```
Comment (jow-) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js]: You can directly use `value` here. The children filter thing is only needed when you want to lookup the values of other options.

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js [hunk 2]
```diff
@@ -1509,11 +1526,28 @@ return L.view.extend({
...
+					o = ss.taboption('encryption', form.Flag, 'ca_cert2_usesystem', _('Use system certificates for inner-tunnel'), _("Validate server certificate using built-in system CA bundle,<br />requires the \"ca-bundle\" package"))
+					o.enabled = '1';
+					o.disabled = '0';
+					o.default = o.disabled;
 					o.depends({ mode: 'sta', auth: 'EAP-TLS', encryption: 'wpa' });
 					o.depends({ mode: 'sta', auth: 'EAP-TLS', encryption: 'wpa2' });
 					o.depends({ mode: 'sta-wds', auth: 'EAP-TLS', encryption: 'wpa' });
 					o.depends({ mode: 'sta-wds', auth: 'EAP-TLS', encryption: 'wpa2' });
+					o.validate = function(section_id, value) {
+						var eo = this.section.children.filter(function(o) { return o.option == 'ca_cert2_usesystem' })[0],
```
Comment (jow-) [inline modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js]: You can directly use `value` here. The children filter thing is only needed when you want to lookup the values of other options.

File: modules/luci-base/root/usr/libexec/rpcd/luci [hunk 3]
```diff
@@ -216,6 +216,7 @@ local methods = {
 			rv.sysntpd       = fs.readlink("/usr/sbin/ntpd") and true
 			rv.ipv6          = fs.access("/proc/net/ipv6_route")
 			rv.dropbear      = fs.access("/usr/sbin/dropbear")
+			rv.cabundle      = fs.access("/etc/ssl/certs/ca-certificates.crt")
```
Comment (jow-) [inline modules/luci-base/root/usr/libexec/rpcd/luci]: Can fold this out into a separate commit?

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js (hunk 1 above)
Comment (swg0101) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js]: Interesting, I tried value before and it returned null. Let me try that again.

File: modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js (hunk 2 above)
Comment (swg0101) [reply modules/luci-mod-network/htdocs/luci-static/resources/view/network/wireless.js]: Interesting, I tried value before and it returned null. Let me try that again.

File: modules/luci-base/root/usr/libexec/rpcd/luci (hunk 3 above)
Comment (swg0101) [reply modules/luci-base/root/usr/libexec/rpcd/luci]: Sure

**Comment by jow-** [comment]
Merged, thank you.

**Pull request merged**